packages/catalog-client/src/CatalogClient.ts

```diff
@@ -168,6 +176,51 @@ export class CatalogClient implements CatalogApi {
     return { items: entities.sort(refCompare) };
   }
 
+  /**
+   * Implements getEntities in terms of smaller queryEntities calls, to avoid
+   * overloading the server with massive queries.
+   */
+  private async emulateGetEntitiesWithQueryEntities(
+    initialRequest: QueryEntitiesInitialRequest,
+    options?: CatalogRequestOptions,
+  ): Promise<GetEntitiesResponse> {
+    const items: Entity[] = [];
+    const maxItemsPerRequest = 1000; // TODO(freben): Make this not hardcoded?
+
+    let request: QueryEntitiesRequest = {
+      ...initialRequest,
+      limit:
+        initialRequest.limit !== undefined
+          ? Math.min(initialRequest.limit, maxItemsPerRequest)
+          : maxItemsPerRequest,
+    };
+
+    for (;;) {
```

Comment:
is there a reason to use for instead of while?

Reply:
Matter of taste - this seems to be the more canonical thing to do because it explicitly lacks a condition / value to test at all.